taking-turns.js

```diff
@@ -0,0 +1,209 @@
+    numGuesses++;   // increment number of guesses
+
+    // make sure user guesses a number
+    while (typeof(userGuess) !== 'number') {
+      console.log(`Enter a number please!`);
+      userGuess = parseInt(await ask(`Make a guess! `));
+    }
+
+    // make sure user guesses within the range
+    while (userGuess > currHigh || userGuess < currLow) {
+      if (userGuess > currHigh) console.log(`Hey, I already told you that my number is lower than ${currHigh + 1}!`);
+      if (userGuess < currLow) console.log(`Hey, I already told you my number is higher than ${currLow - 1}`);
+      userGuess = parseInt(await ask(`Make a guess! `));
+    }
+
+    // if user guesses number correctly
+    if (userGuess == compNum) {
+      found = true;
+      console.log(`You guessed it in ${numGuesses} guesses!`);
+    } else if (userGuess > compNum) {   // if userGuess is greater than the computer's number
+      console.log(`You guessed ${userGuess}, that's WRONG!`);
+      console.log(`My number is LOWER than your guess`);
+      currHigh = userGuess - 1;   // subtract one from userGuess to get the updated highest possible value
+      console.log("currHigh: ", currHigh);
+    } else if (userGuess < compNum) {      // else userGuess is less than the computer's number
+      console.log(`You guessed ${userGuess}, that's WRONG!`)
+      console.log(`My number is HIGHER than your guess`);
+      currLow = userGuess + 1;    // add one to userGuess to get the updated lowest possible value
+      console.log("currLow: ", currLow);
+    } else {
+      console.log("That's not a number!");
+      numGuesses--;   // decrement number of guesses to account for bad input
+    }
+    
+  }
+
+  //process.exit();
+}
+
+// main game loop
+async function main() {
+  let keepGoing = true;
+  while (keepGoing) {
+    console.log(`computer turn to guess`);
+    await computerGuesses();
+    if ((await ask(`Keep going? `)).includes('n')) {
+      keepGoing = false;
+      break;
+    }
+    console.log(`user turn to guess`);
+    await userGuesses();
+    if ((await ask(`Keep going? `)).includes('n')) {
+      keepGoing = false;
+      break;
+    }
+  }
+  process.exit();
+}
+
+// helper functions
+
+// function takes a low and a high number and returns the median number
+function findMid(low, high) {
+  if (high > low)
+    return Math.floor((high + low) / 2);
+  else
+    return high;
+}
+
+// function to get a random integer in a range from min to max (inclusive)
+function randomInteger(min, max) {
+  let range = max - min + 1;
+  return min + Math.floor(Math.random() * range);
+}
+
+// error checking for user input, don't allow anything but N or Y
+async function validateInput(yesOrNo, currGuess) {
+  while (yesOrNo.toUpperCase() !== "N" && yesOrNo.toUpperCase() !== "Y") {
+    console.log(`That's not an answer! Tell me Y or N`);
+    yesOrNo = await ask(`Is it... ${currGuess}? `);
+  }
+  return yesOrNo;
+}
+
+main();
```

Comment:
everything looks good, so far. All that is left is to implement a guessing game in which the user guesses the computer's number and an option to set the range of numbers with argv or an await ask.